tests/test_unit/test_napari_plugin/test_data_loader_widget.py

```diff
@@ -302,10 +340,120 @@ def test_dimension_slider_multiple_files(
 
     # Check the maximum number of frames is the number of frames
     # in the longest dataset
-    _, ds_long = request.getfixturevalue("valid_poses_dataset_long")
+    _, ds_long = request.getfixturevalue("valid_poses_path_and_ds")
     assert max_frames == ds_long.sizes["time"]
 
 
+@pytest.mark.parametrize(
+    "list_input_data_files",
+    [
+        [
+            "valid_poses_path_and_ds",
+            "valid_poses_path_and_ds_nan_start",
+        ],  # one with NaNs at the start remains after deletion
+        [
+            "valid_poses_path_and_ds",
+            "valid_poses_path_and_ds_nan_end",
+        ],  # one with NaNs at the end remains after deletion
+        [
+            "valid_poses_path_and_ds",
+            "valid_poses_path_and_ds_nan_start",
+            "valid_poses_path_and_ds_nan_end",
+        ],  # two remain after deletion, with NaNs at the start and end
+        [
+            "valid_poses_path_and_ds",
+            "valid_poses_path_and_ds_short",
+            "valid_poses_path_and_ds_nan_start",
+        ],  # two remain after deletion, the longest one with NaNs at the start
+        [
+            "valid_poses_path_and_ds",
+            "valid_poses_path_and_ds_short",
+            "valid_poses_path_and_ds_nan_end",
+        ],  # two remain after deletion, the longest one with NaNs at the end
+    ],
+)
+def test_dimension_slider_with_deletion(
+    list_input_data_files,
+    make_napari_viewer_proxy,
+    request,
+):
+    """Test that the dimension slider is set to the correct range of frames
+    when loading two point layers, deleting one, and the remaining layer(s)
+    have all NaN values at the start or end.
+    """
+    # Get the input data to load (paths and ds)
+    list_paths, list_datasets = [
+        [
+            request.getfixturevalue(file_name)[j]
+            for file_name in list_input_data_files
+        ]
+        for j in range(2)
+    ]
+
+    # Check the expected number of datasets have NaN values
+    # at the start or end
+    expected_datasets_with_nans = sum(
+        ["nan" in file_name for file_name in list_input_data_files]
+    )
+    actual_datasets_with_nans = sum(
+        [
+            any(
+                [
+                    ds.position.sel(time=ds.coords["time"][0])
+                    .isnull()
+                    .all()
+                    .values,
+                    ds.position.sel(time=ds.coords["time"][-1])
+                    .isnull()
+                    .all()
+                    .values,
+                ]
+            )
+            for ds in list_datasets
+        ]
+    )
+    assert actual_datasets_with_nans == expected_datasets_with_nans
+
+    # Get the maximum number of frames from all datasets
+    max_frames = max(ds.sizes["time"] for ds in list_datasets)
```

Comment:
this particular variable is never used, as this same line repeates itself a few lines down.

Reply:
good catch, I think I actually missed an additional assertion I wanted to have in between - I added it.